src/test/java/org/apache/maven/plugins/stage/RepositoryCopierTest.java

```diff
@@ -19,46 +19,150 @@
  * under the License.
  */
 
-import org.codehaus.plexus.PlexusTestCase;
-import org.codehaus.plexus.util.FileUtils;
-import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Reader;
+import com.jcraft.jsch.JSch;
+import org.apache.commons.io.FileUtils;
+import org.apache.maven.artifact.manager.WagonManager;
 import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Reader;
 import org.apache.maven.wagon.repository.Repository;
+import org.apache.sshd.common.NamedFactory;
+import org.apache.sshd.common.file.virtualfs.VirtualFileSystemFactory;
+import org.apache.sshd.common.kex.KeyExchange;
+import org.apache.sshd.common.keyprovider.AbstractFileKeyPairProvider;
+import org.apache.sshd.common.session.Session;
+import org.apache.sshd.common.session.SessionListener;
+import org.apache.sshd.common.util.SecurityUtils;
+import org.apache.sshd.server.Command;
+import org.apache.sshd.server.SshServer;
+import org.apache.sshd.server.auth.UserAuth;
+import org.apache.sshd.server.auth.UserAuthNoneFactory;
+import org.apache.sshd.server.auth.pubkey.AcceptAllPublickeyAuthenticator;
+import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
+import org.apache.sshd.server.scp.ScpCommandFactory;
+import org.apache.sshd.server.subsystem.sftp.SftpSubsystemFactory;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.codehaus.plexus.PlexusTestCase;
 
 import java.io.File;
-import java.io.Reader;
 import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Path;
+import java.security.Security;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 /** @author Jason van Zyl */
 public class RepositoryCopierTest
     extends PlexusTestCase
 {
-    private String version = "2.0.6";
+    private static final int PORT = 3543;
+    private static final String STAGING_REPOSITORY = "src/test/staging-repository";
+    private static final String TARGET_REPOSITORY = "src/test/target-repository";
+    private static final String WORKING_TARGET_REPOSITORY = "src/test/working-target-repository";
```

Comment:
I'm curious why only this one is added to .gitignore. Also, if it's a temporary directory perhaps use `Files.createTempDirectory` instead of gitignoring

Reply:
You're right, temp dirs are a better solution. I've changed it.